lib/state/action/pokemon_actions.dart

```diff
@@ -21,6 +22,26 @@ class GetPokemonsAction extends LoadingAction {
   }
 }
 
+///  Searching Pokemon Action
+class SearchPokemonsAction extends ReduxAction<AppState> {
+  SearchPokemonsAction({required this.queryText});
+
+  final String queryText;
+
+  @override
+  AppState reduce() {
+    final searchedPokemons =
+        state.pokemons.filter((pokemon) => pokemon.name.contains(queryText.toLowerCase())).toList();
+    return state.copyWith(searchedPokemons: searchedPokemons);
+  }
+}
+
+/// Clearing Searched Pokemon on the store
```

Comment:
should be `in` not `on` 🙂 
also don't capitalize Searched Pokemon, use `[searchedPokemons]` instead 🙂